### PR TITLE
Fixed discussion user sync issue for users with no profile

### PR DESCRIPTION
**Pull request by rhysyngsun**
#### What are the relevant tickets?
N/A

#### What's this PR do?
This fixes edge cases where a user has no profile. This would only happen for admin users or simialr edge cases.

#### How should this be manually tested?
Tests should pass

### CODE REVIEW

File: discussions/api.py [hunk 1]
```diff
@@ -90,7 +90,10 @@ def create_discussion_user(discussion_user):
     Raises:
         DiscussionUserSyncException: if there was an error syncing the profile
     """
-    profile = discussion_user.user.profile
+    try:
```
Comment (noisecapella) [inline discussions/api.py]: It seems strange to me to have the code raise a `DiscussionUserSyncException` if the API call fails but silence this error. It seems like we would want to raise an error since it's an error case, we just don't want it to halt the processing in the task or fill sentry with useless errors. What do you think of filtering out the missing profile and missing discussion username cases in `get_membership_ids` instead?

**Comment by pdpinch** [comment]
https://sentry.io/mit-office-of-digital-learning/micromasters/issues/396062807/

File: discussions/api.py (hunk 1 above)
Comment (rhysyngsun) [reply discussions/api.py]: @noisecapella I added code to filter out in `get_membership_ids`, the missing username should no longer be a case now too.
